src/hooks/useNoSignInRedirect.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+import useBoundStore from "../store/client/useBoundStore";
+import { getAuth } from "firebase/auth";
+
+const useNoSignInRedirect = () => {
+  const navigate = useNavigate();
+  const isSignIn = useBoundStore((state) => state.isSignIn);
+
+  const auth = getAuth();
+  const user = auth.currentUser;
+
+  useEffect(() => {
+    if (!isSignIn || !user) {
+      navigate("/");
+    }
+  }, [isSignIn, navigate, user]);
```

Comment:
```suggestion
if (!isSignIn || !user) {
   navigate("/");
}
```
커스텀훅 내 불필요한 Effect 일 것 같아, 검토를 부탁 드립니다!
아래와 같이 로그인 여부를 검사할 페이지 단위 컴포넌트에서만 호출하고 있어서, 커스텀훅 내에서 if로 분기처리만 되면 될 것 같은데 어떻게 생각하시나요? :)
```js
const MyPage = () => {
  useNoSignInRedirect(); // 커스텀훅 호출부

  return <div className="mx-auto mt-80 w-full max-w-1200 px-10">마이페이지</div>;
};
```
- [You Might Not Need an Effect](https://react.dev/learn/you-might-not-need-an-effect)

Reply:
![image](https://github.com/user-attachments/assets/9e5c8c3f-9a11-47cf-8877-e2bd3ef60bb3)

@jin-ttao navigate 함수는 컴포넌트가 마운트되고 실행되어져야 한다네요..! `useEffect`를 빼면 컴포넌트가 처음 렌더링되는 즉시 호출되고 있으므로 `useEffect`를 써서 `useEffect` 훅에 의해 트리거 되도록 하였습니다.

Reply:
그렇군요! 현재 로그인이 안되어서 검증을 못할 것이라 생각했는데, 다음에는 검증까지 최대한 해볼게요! 현재 방식을 유지하는 것이 안전할 것 같습니다 :)